src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.cpp

```diff
@@ -142,7 +145,7 @@ bool MultipleAnalogSensorsClient::close()
 
 size_t MultipleAnalogSensorsClient::genericGetNrOfSensors(const std::vector<SensorMetadata>& metadataVector) const
 {
-    return metadataVector.size();
+    return m_externalConnection ? 0 : metadataVector.size();
```

Comment:
I think this at least the getSensors call can be supported, with an implementation like the one proposed. 
```suggestion
    if (!m_externalConnection) { 
        return metadataVector.size();
    } else {
        std::lock_guard<std::mutex>  guard(m_streamingPort.dataMutex);
        m_streamingPort.updateTimeoutStatus();
        if (m_streamingPort.status == yarp::dev::MAS_OK)  {
            return measurementsVector.measurements.size()
        } else {
            return 0;
        }
    }    
```
But we need to add `measurementsVector` as an argument.